File: services/notes-service/src/main/java/org/webber/notesservice/GlobalExceptionHandler.java
package org.webber.notesservice;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler({InternalError.class, Exception.class})
    public ResponseEntity<String> handleServerException(Exception ex) {
        return new ResponseEntity<>("Error on the server side", HttpStatus.INTERNAL_SERVER_ERROR);
    }
}

File: services/flashcards-service/src/main/kotlin/org/webber/flashcardsservice/GlobalExceptionHandler.kt
package org.webber.flashcardsservice

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.webber.flashcardsservice.exceptions.ServiceException

@ControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(ServiceException::class)
    fun handleServiceException(e: ServiceException): ResponseEntity<String> {
        return ResponseEntity(e.message, e.statusCode)
    }

    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception): ResponseEntity<String> {
        return ResponseEntity("Internal server error", HttpStatus.INTERNAL_SERVER_ERROR)
    }
}